### DISCUSSION TITLE
New API image filenames

### DISCUSSION THREAD

**Discussion by slightlyfaulty** [dicebear/dicebear#224] [Ideas]
With the new API (#179), if you save an image the filename given by default is the last segment in the path (at least in Chrome). For example:

```
https://api.dicebear.com/4.x/avataaars/png?seed=jane&size=32&b=red --> png.png
https://api.dicebear.com/4.x/avataaars/png/seed=jane&size=32&b=red --> seed=jane&size=32&b=red.png
```

The old/current API works well in this regard because the last segment in the URL is a filename:

```
https://avatars.dicebear.com/api/avataaars/jane.png?size=32&b=red --> jane.png
```

Could the new API output a nicer filename, or is it not possible to add HTTP headers to the CDN file?

```http
content-disposition: inline; filename="<seed>.<format>"
```

Alternatively, the seed could always be the last segment of the URL:

```
https://api.dicebear.com/4.x/avataaars/png/jane?size=32&b=red --> jane.png
https://api.dicebear.com/4.x/avataaars/png/size=32&b=red/jane --> jane.png
```

But perhaps the tradeoffs of this simply aren't worth the minor benefit of having nicer filenames 🤷‍♂️

**Comment by FlorianKoerner** [comment]
Hi @slightlyfaulty

That's a really good idea. 👍  I have built it in directly.
https://github.com/dicebear/api/commit/3c0bb4c0654bc761d1eb17d2405340cf14bb4963

**Comment by FlorianKoerner** [comment]
And crashed the API directly with it, I see. I' m on it.

**Reply by FlorianKoerner** [reply]
Fixed

**Comment by slightlyfaulty** [comment]
Very nice! I see that the filename is the `style` instead of the `seed`. Not sure if this is intended, but it actually makes more sense in cases where seeds might be convoluted, arbitrary strings, or even blank.

It also doesn't work when used as Gravatar default images unfortunately, since it seems they don't forward the header when serving the image. Not a huge deal though in any case.

**Reply by FlorianKoerner** [reply]
> I see that the filename is the style instead of the seed. Not sure if this is intended

It is intended by me that the style name is used as filename. I had a combination before (avatar style + seed), but then some seed crashed the API. - And this although I had checked whether the seed can be a valid file name.

In the end, I prefer to take the safe way and not use user input as a filename.

> It also doesn't work when used as Gravatar default images unfortunately, since it seems they don't forward the header when serving the image. Not a huge deal though in any case.

Too bad it doesn't work with Gravatar. - But also a rare use case.

**Reply by slightlyfaulty** [reply]
Makes sense. Thanks again for the responsiveness @FlorianKoerner!
